src/helpers.rs

```diff
@@ -324,6 +324,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         bug!("No field named {} in type {}", name, base.layout().ty);
     }
 
+    fn try_project_field_named<P: Projectable<'tcx, Provenance>>(
```

Comment:
Please don't duplicate this code. Instead, use this function to implement `project_field_named` and `projectable_has_field`.